### PR TITLE
Values are represented as strings in FL_OCLINT_SELECT_REGEX

### LINKED ISSUE #10069: oclint - How can I define `FL_OCLINT_SELECT_REGEX` value in a `.env` file
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

How can I define `FL_OCLINT_SELECT_REGEX` value in a `.env` file? If try to use `FL_OCLINT_SELECT_REGEX=/src/` then I get `type mismatch: String given`.

Same error if I try to use:
```
FL_OCLINT_SELECT_REGEX=/src/
FL_OCLINT_SELECT_REGEX=`/src/`
FL_OCLINT_SELECT_REGEX=\/src\/
FL_OCLINT_SELECT_REGEX=\\/src\\/
```

##### Complete output when running fastlane, including the stack trace and command used

```
10:25 user:MyProject$ fastlane ios run_analyzer --verbose
[10:25:16]: For more information, check out https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
DEBUG [2017-08-17 10:25:19.07]: Checking if there are any plugins that should be loaded...
INFO [2017-08-17 10:25:19.19]: gem 'spaceship' is already installed
INFO [2017-08-17 10:25:19.19]: -------------------------------------------------
INFO [2017-08-17 10:25:19.19]: --- Step: Verifying required fastlane version ---
INFO [2017-08-17 10:25:19.20]: -------------------------------------------------
INFO [2017-08-17 10:25:19.20]: Your fastlane version 2.53.1 matches the minimum requirement of 2.50.0  ✅
INFO [2017-08-17 10:25:19.20]: ------------------------------
INFO [2017-08-17 10:25:19.20]: --- Step: default_platform ---
INFO [2017-08-17 10:25:19.20]: ------------------------------
INFO [2017-08-17 10:25:19.20]: Driving the lane 'ios run_analyzer' 🚀
INFO [2017-08-17 10:25:19.20]: Lane run_analyzer started.
INFO [2017-08-17 10:25:19.20]: --------------------
INFO [2017-08-17 10:25:19.20]: --- Step: oclint ---
INFO [2017-08-17 10:25:19.20]: --------------------
DEBUG [2017-08-17 10:25:19.21]: Taking value for 'compile_commands' from environment variable 'FL_OCLINT_COMPILE_COMMANDS'
DEBUG [2017-08-17 10:25:19.21]: Taking value for 'compile_commands' from environment variable 'FL_OCLINT_COMPILE_COMMANDS'
DEBUG [2017-08-17 10:25:19.21]: Taking value for 'select_regex' from environment variable 'FL_OCLINT_SELECT_REGEX'
DEBUG [2017-08-17 10:25:19.21]: Taking value for 'select_regex' from environment variable 'FL_OCLINT_SELECT_REGEX'
INFO [2017-08-17 10:25:24.98]: Lane run_analyzer failed to complete.
WARN [2017-08-17 10:25:24.98]: Lane Context:
INFO [2017-08-17 10:25:24.98]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios run_analyzer"}
ERROR [2017-08-17 10:25:24.98]: type mismatch: String given
INFO [2017-08-17 10:25:24.98]: Successfully generated documentation at path '/Users/user/projects/ios/project/MyProject/fastlane/README.md'

+------+------------------------------+-------------+
|                 fastlane summary                  |
+------+------------------------------+-------------+
| Step | Action                       | Time (in s) |
+------+------------------------------+-------------+
| 1    | Verifying required fastlane  | 0           |
|      | version                      |             |
| 2    | default_platform             | 0           |
| 💥   | oclint                       | 0           |
+------+------------------------------+-------------+

ERROR [2017-08-17 10:25:24.99]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...
Search query: type mismatch: String given

URL: https://api.github.com/search/issues?q=type%20mismatch:%20String%20given+repo:fastlane/fastlane
Found no similar issues. To create a new issue, please visit:
https://github.com/fastlane/fastlane/issues/new
Run `fastlane env` to append the fastlane environment to your issue
/Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/actions/oclint.rb:45:in `=~': [!] type mismatch: String given (TypeError)
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/actions/oclint.rb:45:in `block in run'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/actions/oclint.rb:42:in `select!'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/actions/oclint.rb:42:in `run'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/runner.rb:252:in `block (2 levels) in execute_action'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/runner.rb:230:in `block in execute_action'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/runner.rb:226:in `chdir'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/runner.rb:226:in `execute_action'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/runner.rb:148:in `trigger_action_by_name'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
	from Fastfile:140:in `block (2 levels) in parsing_binding'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/lane_manager.rb:52:in `cruise_lane'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/command_line_handler.rb:30:in `handle'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/commands_generator.rb:104:in `block (2 levels) in run'
	from /Users/user/.gem/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
	from /Users/user/.gem/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
	from /Users/user/.gem/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:64:in `run!'
	from /Users/user/.gem/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/commands_generator.rb:303:in `run'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	from /Users/user/.gem/gems/fastlane-2.53.1/fastlane/lib/fastlane/cli_tools_distributor.rb:66:in `take_off'
	from /Users/user/.gem/gems/fastlane-2.53.1/bin/fastlane:20:in `<top (required)>'
	from /Users/user/.gem/bin/fastlane:23:in `load'
	from /Users/user/.gem/bin/fastlane:23:in `<main>'

```

### PR DESCRIPTION
**Pull request by taquitos**
Need to make them regexs, should fix #10069

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
Fastlane Fastlane::FastFile OCLint Integration works with select regex when regex is string | <a href='https://github.com/fastlane/fastlane/blob/b5925bf00a626d1f1df8e81a209a52f037c27c75/./fastlane/spec/actions_specs/oclint_spec.rb'>oclint_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Review by endocrimes — CHANGES_REQUESTED** [review]

File: fastlane/lib/fastlane/actions/oclint.rb [hunk 1]
```diff
@@ -90,6 +92,22 @@ def self.run(params)
         return Action.sh(command)
       end
 
+      # return a proper regex object if regex string is single-quoted
+      def self.ensure_regex_is_not_string!(regex)
+        unless regex.nil?
+          if regex.kind_of?(String)
+            safer_eval = Thread.new do
+              $SAFE = 1
```
Comment (snatchev) [inline fastlane/lib/fastlane/actions/oclint.rb]: ++ for `Regexp.new`. Will probably have to strip the preceding and ending `/`

Comment (taquitos) [inline fastlane/lib/fastlane/actions/oclint.rb]: Regexp requires that the string be single-quoted, otherwise you have to extra-escape things.

**Comment by taquitos** [comment]
@DanToml + @snatchev any thoughts on a better way to ensure that strings which are double-quoted, as well as single-quoted are turned into Rexexps?

**Comment by snatchev** [comment]
@taquitos could you explain the desired behavior? AFAIK it's not possible to determine if a string literal was created via single or double quotes (or `%q{}` / `%Q{}` for that matter).

**Comment by snatchev** [comment]
If the goal is to turn a specific input into a regex v.s. literal string match, I propose something like this:

```ruby
if select_regex =~ %r{^/(.*)/$}
  select_regex = Regexp.new($1)
else
  select_regex = Regexp.new(Regexp.escape(select_regex))
end
```

**Comment by taquitos** [comment]
@snatchev, in the referenced issue, setting `FL_OCLINT_SELECT_REGEX` to a string causes issues when passing the string through Regexp.new(). The issue is that depending on single vs double quoted, you may have to escape `/` . My thought was: allow whatever as a string, and don't worry about escapes because eval() will turn it into a regex. That way parsing single quotes or double quotes doesn't matter, people can just write the regex in a string without worrying which means fewer github issues. That being said, I'm starting to think we should just stick to Regexp and tell people to escape their regexes if needed.

**Comment by snatchev** [comment]
@taquitos Sounds good. I agree that we shouldn't try to be too clever here.

In that case, we can simplify some of the logic in this method
```ruby
def self.ensure_regex_is_not_string!(regex)
  return regex unless regex.kind_of?(String)

  Regexp.new(regex)
end
```

**Review by endocrimes — APPROVED** [review]

**Comment by taquitos** [comment]
Woo, thanks for the review @DanToml!

**Pull request merged**